src/components/OptionListContextProvider.tsx

```diff
@@ -283,7 +291,7 @@ const useOptionsList = (options?: {shouldInitialize: boolean}) => {
             return;
         }
         setInternalOptions(optionsList);
```

Comment:
```suggestion
        setInternalOptions(optionsList);
        setAreInternalOptionsInitialized(areOptionsInitialized);
```

Same suggestion for the other `setInternalOptions()`.